driver-mapping/src/main/java/com/datastax/driver/mapping/configuration/naming/NamingStrategy.java

```diff
@@ -0,0 +1,130 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.mapping.configuration.naming;
+
+public class NamingStrategy {
+
+    private boolean enabled;
+
+    private NamingConvention javaConvention;
+
+    private NamingConvention cassandraConvention;
+
+    public NamingStrategy(NamingConvention javaConvention, NamingConvention cassandraConvention) {
+        setJavaConvention(javaConvention);
+        setCassandraConvention(cassandraConvention);
+        enable();
+    }
+
+    public NamingStrategy(NamingStrategy toCopy) {
+        this.enabled = toCopy.enabled;
+        this.javaConvention = toCopy.javaConvention;
+        this.cassandraConvention = toCopy.cassandraConvention;
+    }
+
+    /**
+     * Receive a String input assumed to be in the Java naming convention configured,
+     * returns the String translation to the Cassandra naming convention.
+     *
+     * @param input value to translate
+     * @return Cassandra naming convention translation of the input
+     */
+    public String toCassandra(String input) {
+        if (!enabled || javaConvention.equals(cassandraConvention)) {
+            return input;
+        }
+        return cassandraConvention.join(javaConvention.split(input));
+    }
+
+    /**
+     * Receive a String input assumed to be in the Cassandra naming convention configured,
+     * returns the String translation to the Java naming convention.
+     *
+     * @param input value to translate
+     * @return Java naming convention translation of the input
+     */
+    public String toJava(String input) {
```

Comment:
I went for designing an independent interface for handling translation of naming conventions to it seemed trivial to implement both ways. If we prefer to avoid unused code we can remove it. If in the future we find a need to translate Cassandra column names to java fields, as you can see, the implementation is 3 lines.
So i think we can go for removing right now.